### ISSUE TITLE
Screen reader: loosing cursor in Suggest widget

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#4351]
- VSCode Version: 0.10.12-alpha
- OS Version: Windows 10 / NVDA screen reader

Steps to Reproduce:
1. enable NVDA screen reader
2. open a TS file
3. open code complete for something were you get one entry
4. press ALT+Down
5. press escape to dismiss the dialog
6. press cursor up.

Observer: the cursor now disappears. You need to click into the editor to get the cursor back.

**Comment by alexandrudima** [comment]
@dbaeumer NVDA supports different modes. I think you have entered an object navigation mode of NVDA. I use `Insert+Space` to bring back NVDA into text input mode.

**Comment by alexandrudima** [comment]
@zersiax 

From reading a bit the [NVDA user guide](http://www.nvaccess.org/files/nvda/documentation/userGuide.html), it looks like Escape switches back to browse mode if focus mode was previously switched to automatically.

We use `Escape` in many places in VSCode to dismiss or close things, including closing the suggestions list, dismissing messages, closing the find widget, etc.

Should we bind out of the box both `Escape` and `Alt+Escape` for these commands in order to have a way to dismiss/close things that does not collide with NVDA?

**Comment by alexandrudima** [comment]
Happy to hear that's not a general problem.

@dbaeumer ran into a case with NVDA where dismissing the suggestion list with `Escape` brought NVDA to browse mode.

I think that's because he clicked in the editor, which brought NVDA automatically to focus mode. Then, when pressing `Escape` to dismiss the suggestion list, NVDA switched back to browse mode (as it entered the focus mode automatically).

We then chatted about it and thought that maybe this happens all the time, but we were not sure, that's why we reached out to you. Thank you for confirming this is not a general problem.

**Comment by zersiax** [comment]
Wait wait wait I am not saying that :)
Let me do some testing real quick :)

**Comment by zersiax** [comment]
No, you are right. I am seeing the same behavior. Escaping out of the
suggestion list doesn't actually dismiss the suggestion list but takes
you to browse mode. So I think thesekeys are necessary after all.

**Comment by alexandrudima** [comment]
Sorry for jumping to conclusions, I guess I read what I want to read :).

I'll go through the places we use Escape and also add Alt+Escape as an alternative keybinding doing the same.

**Comment by alexandrudima** [comment]
I ended up adding Shift+Escape as an alternative, since Alt+Escape is some sort of equivalent to Alt+Tab on Windows

**Comment by dbaeumer** [comment]
@alexandrudima while verifying this I discovered the following problems:

- Alt cursor down now moves the line in the editor and doesn't select the next entry in the list anymore
- Shift Escape doesn't close the code complete window.

**Comment by dbaeumer** [comment]
And nothing thing I noticed. When using arrow down and up in the code complete window to navigate and closing the windows with escape NVDA doesn't get into 'Browsing' mode. It says in focus mode. So may be the Alt+Down / Up triggers something as well which puts NVDA in browsing mode.

**Comment by dbaeumer** [comment]
@alexandrudima hold back. It looks like the insider build didn't update for me. I will close and retest.

**Comment by dbaeumer** [comment]
OK. Retested on latest insider from today and here is what I see:

- Alt+Up/Down works as expected in code complete
- Shift+Escape closes the code complete Window, but NVDA still ends in Browsing mode (no focus in the editor anymore). 

What is interesting is: when using arrow down and up in the code complete window to navigate and closing the windows with escape or shift escape NVDA DOESNOT get into 'Browsing' mode. It stays in focus mode. So may be the Alt+Down / Up triggers something which puts NVDA in browsing mode.

**Comment by alexandrudima** [comment]
@dbaeumer Let's look into it together. NVDA does not go into 'Browsing mode' for me when I use alt+down, alt+up and shift+escape.

**Comment by dbaeumer** [comment]
@alexandrudima makes sense.

**Comment by dbaeumer** [comment]
@egamma I think we should reopen this and target for April. The makes sense comment was meant to 'Let's look into it together'. I still see the issue on my machine.

**Comment by egamma** [comment]
:+1:

**Comment by alexandrudima** [comment]
@dbaeumer Here is the recommended flow that works for me:

* in the suggest widget, press arrow down and arrow up to move between suggestions
* to dismiss the suggest widget, press Shift+Escape (and not Escape since Escape leaves focus mode in NVDA)

I agree, alt+up and alt+down with the suggest window opened cause NVDA to leave focus mode. But then, arrow down and arrow up can be used instead.

**Comment by alexandrudima** [comment]
I take it back, using the arrow down/arrow up keys reads the whole line again. I'll look into adding ctrl+arrows instead of alt+arrows for suggestion list navigation to circumvent this.

**Comment by chrmarti** [comment]
Can't verify using a VM on Mac due to Ctrl+Up/Down being bound by the OS and going through the VM software's keyboard tool trigger inconsistent behavior. I suggest someone with a Windows machine to verify.

**Comment by roblourens** [comment]
I can verify that ctrl+up/down goes through the list and reads the selected suggestion, and that shift+escape hides the suggestion box without nvda leaving focus mode.
